### ISSUE TITLE
[iOS] Exception clicking Cancel on Keyboards screen (iPad Pro 12.9, 2nd gen)

### ISSUE THREAD

**Issue by tombogle** [keymanapp/keyman#1046]
I got this while running a locally built version of Keyman with the debugger, but my code base should be identical to the existing 10.0 release. Keyboard display was messed up (see #1045). I clicked the globe button to go into Keyboards screen, and then immediately clicked Cancel. Full output from debugging window follows:

```
2018-07-02 15:56:34.337867-0400 Keyman[4431:232735] [ApplicationLifecycle] Windows were created before application initialzation completed. This may result in incorrect visual appearance.
2018-07-02 15:56:34.338871-0400 Keyman[4431:232735] +[CATransaction synchronize] called within transaction
2018-07-02 15:56:34.427 [Info] > Keyman Version: 10.0.0 Build: 10.0.0 PID: 4431
2018-07-02 15:56:34.427 [Info] > XCGLogger Version: 6.0.2 - Level: Debug
2018-07-02 15:56:34.461 [Info] [Migrations.swift:38] migrateUserDefaultsToStructs(storage:) > User keyboard list or current keyboard missing. Skipping migration.
2018-07-02 15:56:34.466 [Info] [Migrations.swift:132] migrateForKMP(storage:) > Migrating from base directory: file:///Users/tom/Library/Developer/CoreSimulator/Devices/C1F4C6AA-DC4D-454F-91B3-80CB55D04EA7/data/Containers/Shared/AppGroup/EC73FDF4-5FC6-478B-920D-1C75276B0902/keyman
2018-07-02 15:56:34.466 [Info] [Migrations.swift:135] migrateForKMP(storage:) > No user keyboards to migrate
2018-07-02 15:56:34.613 [Info] [APIKeyboardRepository.swift:35] fetch(completionHandler:) > Connecting to Keyman cloud: https://api.keyman.com/cloud/4.0/languages?dateformat=seconds&device=ipad&version=10.0.0&languageidtype=bcp47.
2018-07-02 15:56:34.623 [Info] [FontManager.swift:88] registerFont(at:) > Registered font DejaVuSans at file:///Users/tom/Library/Developer/CoreSimulator/Devices/C1F4C6AA-DC4D-454F-91B3-80CB55D04EA7/data/Containers/Shared/AppGroup/EC73FDF4-5FC6-478B-920D-1C75276B0902/keyman/keyboards/european2/DejaVuSans.ttf
2018-07-02 15:56:34.692 [Debug] [TextView.swift:96] setKeymanDelegate > TextView: 140303965240832 keymanDelegate set to: Optional(<Keyman.MainViewController: 0x7f9b1004d200>)
2018-07-02 15:56:35.051989-0400 Keyman[4431:232735] Reachability Flag Status: -R ------- networkStatusForFlags
2018-07-02 15:56:35.052 [Debug] [Manager.swift:727] reachabilityChanged > Reachability changed to 'Reachable Via WiFi'
2018-07-02 15:56:35.052293-0400 Keyman[4431:232735] Reachability Flag Status: -R ------- networkStatusForFlags
2018-07-02 15:56:35.054 [Debug] [InfoViewController.swift:68] loadFromServer() > Info page URL: http://keyman.com/iphone-and-ipad/app/?active=european2&installed=european2
2018-07-02 15:56:35.054201-0400 Keyman[4431:232735] Reachability Flag Status: -R ------- networkStatusForFlags
2018-07-02 15:56:35.055 [Debug] [InfoViewController.swift:68] loadFromServer() > Info page URL: http://keyman.com/iphone-and-ipad/app/?active=european2&installed=european2
2018-07-02 15:56:35.746137-0400 Keyman[4431:232855] Task <629ABC94-3A58-48D6-87AF-E99EBC02EC3F>.<0> HTTP load failed (error code: -999 [1:89])
2018-07-02 15:56:36.069 [Info] [APIKeyboardRepository.swift:95] apiCompletionHandler(data:response:error:fetchCompletionHandler:) > Request completed -- 928 languages.
2018-07-02 15:56:36.355 [Info] [KeymanWebViewController.swift:286] userContentController(_:didReceive:) > KMW Log: No keyboard stubs exist - cannot initialize keyboard!
2018-07-02 15:56:36.355 [Info] [Manager.swift:1149] keyboardLoaded > Loaded keyboard.
2018-07-02 15:56:36.357 [Info] [Manager.swift:223] setKeyboard > Setting language: <keyboardID: "european2", languageID: "en">
2018-07-02 15:56:36.358 [Debug] [KeymanWebViewController.swift:178] setKeyboard > Keyboard stub: {"KL":"English","KI":"Keyboard_european2","KN":"EuroLatin2 Keyboard","KLC":"en","KOskFont":{"family":"european2__display","files":["file:\/\/\/Users\/tom\/Library\/Developer\/CoreSimulator\/Devices\/C1F4C6AA-DC4D-454F-91B3-80CB55D04EA7\/data\/Containers\/Shared\/AppGroup\/EC73FDF4-5FC6-478B-920D-1C75276B0902\/keyman\/keyboards\/european2\/DejaVuSans.ttf"]},"KFont":{"family":"european2__display","files":["file:\/\/\/Users\/tom\/Library\/Developer\/CoreSimulator\/Devices\/C1F4C6AA-DC4D-454F-91B3-80CB55D04EA7\/data\/Containers\/Shared\/AppGroup\/EC73FDF4-5FC6-478B-920D-1C75276B0902\/keyman\/keyboards\/european2\/DejaVuSans.ttf"]},"KF":"file:\/\/\/Users\/tom\/Library\/Developer\/CoreSimulator\/Devices\/C1F4C6AA-DC4D-454F-91B3-80CB55D04EA7\/data\/Containers\/Shared\/AppGroup\/EC73FDF4-5FC6-478B-920D-1C75276B0902\/keyman\/keyboards\/european2\/european2-1.6.js"}
2018-07-02 15:56:36.361 [Debug] [TextView.swift:149] keyboardChanged > TextView: 140303965240832 setFont: DejaVu Sans
2018-07-02 15:56:36.380288-0400 Keyman[4431:232735] [MC] System group container for systemgroup.com.apple.configurationprofiles path is /Users/tom/Library/Developer/CoreSimulator/Devices/C1F4C6AA-DC4D-454F-91B3-80CB55D04EA7/data/Containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2018-07-02 15:56:36.380709-0400 Keyman[4431:232735] [MC] Reading from private effective user settings.
2018-07-02 15:56:36.446 [Debug] [TextView.swift:291] textViewDidBeginEditing > TextView: 140303965240832 setFont: DejaVu Sans
2018-07-02 15:56:36.447 [Debug] [TextView.swift:296] textViewDidBeginEditing > TextView: 140303965240832 Became first responder. Value: Optional("‎")
2018-07-02 15:56:36.957033-0400 Keyman[4431:232906] WF: === Starting WebFilter logging for process Keyman
2018-07-02 15:56:36.957241-0400 Keyman[4431:232906] WF: _userSettingsForUser : (null)
2018-07-02 15:56:36.957587-0400 Keyman[4431:232906] WF: _WebFilterIsActive returning: NO
2018-07-02 15:59:09.488334-0400 Keyman[4431:232735] *** Terminating app due to uncaught exception 'UIViewControllerHierarchyInconsistency', reason: 'child view controller:<KeymanEngine.KeymanWebViewController: 0x7f9b0fd0a750> should have parent view controller:<UICompatibilityInputViewController: 0x7f9b0fe88fb0> but requested parent is:<UICompatibilityInputViewController: 0x7f9b0fc05e70>'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000107f7d1e6 __exceptionPreprocess + 294
	1   libobjc.A.dylib                     0x0000000107612031 objc_exception_throw + 48
	2   CoreFoundation                      0x0000000107ff2975 +[NSException raise:format:] + 197
	3   UIKit                               0x0000000108cb76eb -[UIViewController _addChildViewController:performHierarchyCheck:notifyWillMove:] + 336
	4   UIKit                               0x0000000108cd992e -[UIViewController(UIContainerViewControllerProtectedMethods) addChildViewController:] + 83
	5   UIKit                               0x00000001094f46b2 +[UICompatibilityInputViewController inputViewControllerWithView:] + 159
	6   UIKit                               0x0000000109300f29 -[UIInputViewSet inputViewController] + 140
	7   UIKit                               0x0000000109300936 -[UIInputViewSet keyboard] + 22
	8   UIKit                               0x00000001092f49f1 -[UIPeripheralHost(UIKitInternal) _reloadInputViewsForResponder:] + 878
	9   UIKit                               0x00000001092fecdb -[UIPeripheralHost(UIKitInternal) _restoreInputViewsWithId:animated:] + 1105
	10  UIKit                               0x0000000108ccb644 -[UIViewController _restoreInputViewsForPresentation] + 269
	11  UIKit                               0x0000000108c8394f -[UIPresentationController runTransitionForCurrentState] + 1533
	12  UIKit                               0x0000000108c81724 -[UIPresentationController _dismissWithAnimationController:interactionController:target:didEndSelector:] + 725
	13  UIKit                               0x0000000108ccb102 -[UIViewController _dismissViewControllerWithAnimationController:interactionController:completion:] + 513
	14  UIKit                               0x0000000108ccad29 -[UIViewController _dismissViewControllerWithTransition:from:completion:] + 634
	15  UIKit                               0x0000000108cca450 -[UIViewController dismissViewControllerWithTransition:completion:] + 1208
	16  UIKit                               0x0000000108cca204 -[UIViewController dismissViewControllerWithTransition:completion:] + 620
	17  UIKit                               0x0000000108cc9958 -[UIViewController _performCoordinatedPresentOrDismiss:animated:] + 511
	18  UIKit                               0x0000000108cccb25 -[UIViewController dismissViewControllerAnimated:completion:] + 136
	19  KeymanEngine                        0x000000010b6220d5 _T012KeymanEngine7ManagerC21dismissKeyboardPickerySo16UIViewControllerCF + 485
	20  KeymanEngine                        0x000000010b6445bd _T012KeymanEngine28KeyboardPickerViewControllerC13cancelClickedyypF + 109
	21  KeymanEngine                        0x000000010b64462c _T012KeymanEngine28KeyboardPickerViewControllerC13cancelClickedyypFTo + 76
	22  UIKit                               0x0000000108b103e8 -[UIApplication sendAction:to:from:forEvent:] + 83
	23  UIKit                               0x0000000109515209 __45-[_UIButtonBarTargetAction _invoke:forEvent:]_block_invoke + 154
	24  UIKit                               0x0000000109515142 -[_UIButtonBarTargetAction _invoke:forEvent:] + 154
	25  UIKit                               0x0000000108b103e8 -[UIApplication sendAction:to:from:forEvent:] + 83
	26  UIKit                               0x0000000108c8b7a4 -[UIControl sendAction:to:forEvent:] + 67
	27  UIKit                               0x0000000108c8bac1 -[UIControl _sendActionsForEvents:withEvent:] + 450
	28  UIKit                               0x0000000108c8aa09 -[UIControl touchesEnded:withEvent:] + 580
	29  UIKit                               0x0000000108b850bf -[UIWindow _sendTouchesForEvent:] + 2729
	30  UIKit                               0x0000000108b867c1 -[UIWindow sendEvent:] + 4086
	31  UIKit                               0x0000000108b2a310 -[UIApplication sendEvent:] + 352
	32  UIKit                               0x000000010946b6af __dispatchPreprocessedEventFromEventQueue + 2796
	33  UIKit                               0x000000010946e2c4 __handleEventQueueInternal + 5949
	34  CoreFoundation                      0x0000000107f1fbb1 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	35  CoreFoundation                      0x0000000107f044af __CFRunLoopDoSources0 + 271
	36  CoreFoundation                      0x0000000107f03a6f __CFRunLoopRun + 1263
	37  CoreFoundation                      0x0000000107f0330b CFRunLoopRunSpecific + 635
	38  GraphicsServices                    0x0000000110f70a73 GSEventRunModal + 62
	39  UIKit                               0x0000000108b0f057 UIApplicationMain + 159
	40  Keyman                              0x0000000106baf087 main + 55
	41  libdyld.dylib                       0x000000010d69f955 start + 1
	42  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) 
```

**Comment by tombogle** [comment]
I have not been able to reproduce on physical device, but it is easy to reproduce in Simulator. Also reproduced on iPad Pro 10.5.

**Comment by tombogle** [comment]
This is "fixed" by the same temporary patch that fixes #1045.
